WordPress/src/main/java/org/wordpress/android/networking/GravatarApi.java

```diff
@@ -76,17 +79,25 @@ public void run() {
                                 if (response.isSuccessful()) {
                                     gravatarUploadListener.onSuccess();
                                 } else {
+                                    AnalyticsTracker.track(AnalyticsTracker.Stat.ME_GRAVATAR_UPLOAD_UNSUCCESSFUL);
+                                    AppLog.w(AppLog.T.API, "Network call unsuccessful trying to upload Gravatar: " +
+                                            response.message());
                                     gravatarUploadListener.onError();
                                 }
                             }
                         });
                     }
 
                     @Override
-                    public void onFailure(okhttp3.Call call, IOException e) {
+                    public void onFailure(okhttp3.Call call, final IOException e) {
                         new Handler(Looper.getMainLooper()).post(new Runnable() {
                             @Override
                             public void run() {
+                                AnalyticsTracker.track(AnalyticsTracker.Stat.ME_GRAVATAR_UPLOAD_EXCEPTION);
+                                CrashlyticsUtils.logException(e, CrashlyticsUtils.ExceptionType.SPECIFIC,
```

Comment:
What if we add a property named `reason` (or something similar), and use it to keep track of the exception message and class?